### PR TITLE
docs(repo): grammar review

### PR DESCRIPTION
**Pull request by heitortsergent**
**Checklist**

- [ ] Tests added / updated
- [x] Docs added / updated

**Additional context**
I took a pass at all the docs using Grammarly to try and fix any grammar mistakes, and also made some small edits where I thought things could be improved.

I was hoping to run the docs through Vale as well using the writing style guide we're using for the Stoplight docs, but I think that would make this PR too big, so I'm going to do this in a separate PR after I work on #2233.

### CODE REVIEW

File: docs/getting-started/2-installation.md [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
 
-For users without Node and/or NPM/Yarn, we provide standalone packages for [all major platforms](https://github.com/stoplightio/spectral/releases). The quickest way to install the appropriate package for your operating system is via this shell script:
+For users without Node.js and/or npm/Yarn, we provide standalone packages for [all major platforms](https://github.com/stoplightio/spectral/releases). The quickest way to install the appropriate package for your operating system is via this shell script:
 
 ```bash
 curl -L https://raw.github.com/stoplightio/spectral/master/scripts/install.sh | sh
 ```
 
-Note, the binaries do _not_ auto-update, so you will need to run it again to install new versions.
+The binaries do _not_ auto-update, so you will need to run it again to install new versions.
```
Comment (pamgoodrich) [inline docs/getting-started/2-installation.md]: Maybe change "it" to "the command"

File: docs/guides/1-workflows.md [hunk 2]
```diff
@@ -50,8 +52,8 @@ See our [CLI documentation](./2-cli.md) to see what other arguments and options
 
 ## Continuous Integration
 
-Spectral can be used in any CI environment that runs run NodeJS or our Docker image: Jenkins, CircleCI, GitHub Actions, etc.
+Spectral can be used in any CI environment that runs Node.js or our Docker image: Jenkins, CircleCI, GitHub Actions, etc.
 
-By enabling the JUnit output format when you lint, most CI servers will show visual results helping people realise what mistakes were made and where.
+By enabling the JUnit output format when you lint, most CI servers will show visual results helping people realize what mistakes were made and where.
```
Comment (pamgoodrich) [inline docs/guides/1-workflows.md]: what mistakes > which mistakes

File: docs/guides/2-cli.md [hunk 3]
```diff
@@ -62,21 +62,21 @@ Here you can build a [custom ruleset](../getting-started/3-rulesets.md), or exte
 
 ## Error Results
 
-Spectral has a few different error severities: `error`, `warn`, `info` and `hint`, and they are in "order" from highest to lowest. By default, all results will be shown regardless of severity, but since v5.0, only the presence of errors will cause a failure status code of 1. Seeing results and getting a failure code for it are now two different things.
+Spectral has a few different error severities: `error`, `warn`, `info`, and `hint`, and they are in "order" from highest to lowest. By default, all results will be shown regardless of severity, but since v5.0, only the presence of errors will cause a failure status code of 1. Seeing results and getting a failure code for it are now two different things.
```
Comment (pamgoodrich) [inline docs/guides/2-cli.md:65]: Do you think we need to keep the reference to v5.0?

File: docs/guides/4-custom-rulesets.md [hunk 4]
```diff
@@ -247,7 +245,7 @@ parserOptions:
 
 Optionally provide a documentation URL to your ruleset in order to help end-users find more information about various warnings. Result messages will sometimes be more than enough to explain what the problem is, but it can also be beneficial to explain _why_ a message exists, and this is a great place to do that.
 
-Whatever you link you provide, the rule name will be appended as an anchor.
+Whatever link you provide, the rule name will be appended as an anchor.
```
Comment (pamgoodrich) [inline docs/guides/4-custom-rulesets.md]: You could simplify this and say "The rule name is appended to the link as an anchor."

File: docs/guides/4-custom-rulesets.md [hunk 5]
```diff
@@ -339,7 +337,7 @@ This code example adds two rules: `valid-rule` and `only-new-json-schema`, thing
 
 For those of you using custom functions, the keywords `functions` & `functionOptions` have been removed, as they were designed to help Spectral find your functions. Now functions are passed as a variable, instead of using a string that contains the name like the JSON/YAML formats.
 
-For now the JSON, YAML, and JS, are all being maintained, and there are no current plans to drop support for any of them.
+For now the JSON, YAML, and JS formats are all being maintained, and there are no current plans to drop support for any of them.
```
Comment (pamgoodrich) [inline docs/guides/4-custom-rulesets.md:340]: "are all being maintained" > "are maintained"

File: docs/guides/5-custom-functions.md [hunk 6]
```diff
@@ -235,14 +233,15 @@ export default createRulesetFunction(
 );
 ```
 
-It's worth keeping in mind, Spectral will attempt to deduplicate messages when they bear the same `code` and target the same `path`.
+It's worth keeping in mind, Spectral will attempt to deduplicate messages when they have the same `code` and target the same `path`.
```
Comment (pamgoodrich) [inline docs/guides/5-custom-functions.md:236]: You could remove "It's worth keeping in mind" to simplify this.

File: docs/guides/5-custom-functions.md [hunk 7]
```diff
@@ -266,8 +265,8 @@ As of Spectral 5.4.0, custom functions can also be asynchronous.
 
 <!-- theme: warning -->
 
-> Ideally linting should always be deterministic, which means if its run 10 times it should return the same results 10 times. To ensure this is the case, please refrain from introducing any logic that is prone to non-deterministic behavior. Examples of this might be contacting external service you have no control over, or that might be unstable, or change the way it responds over time.
-> While, it may seem tempting to have a function that does so, the primary use case is to support libraries that makes async fs calls or exchanging information, i.e. obtaining a dictionary file, with a locally running server, etc.
+> Ideally linting should always be deterministic, which means if it's run 10 times it should return the same results 10 times. To ensure this is the case, please refrain from introducing any logic that is prone to non-deterministic behavior. Examples of this might be contacting an external service you have no control over, or that is unstable, or that changes the way it responds over time.
```
Comment (pamgoodrich) [inline docs/guides/5-custom-functions.md:268]: remove please, please.

File: docs/guides/5-custom-functions.md [hunk 8]
```diff
@@ -266,8 +265,8 @@ As of Spectral 5.4.0, custom functions can also be asynchronous.
...
+> While, it may seem tempting to have a function that does so, the primary use case is to support libraries that makes async fs calls or exchange information, i.e. obtaining a dictionary file, with a locally running server, etc.
```
Comment (pamgoodrich) [inline docs/guides/5-custom-functions.md]: Remove comma after while

File: docs/guides/5-custom-functions.md [hunk 9]
```diff
@@ -332,28 +331,32 @@ This indicates that almost any arbitrary code can be executed.
...
 
-Core functions can be overridden with custom rulesets, so if you'd like to make your own truthy go ahead. Custom functions are only available in the ruleset which defines them, so loading a foo in one ruleset will not clobber a foo in another ruleset.
+Core functions can be overridden with custom rulesets, so if you'd like to make your own `truthy` function you can do so.
+
+Custom functions are only available in the ruleset which defines them, so loading a `foo` function in one ruleset will not affect a `foo` function in another ruleset.
+
+## Performance Tips
 
-## Performance tips
+Try to avoid allocating objects as much as possible if your custom function is very generic, and therefore is expected to be used by plenty of rules.
```
Comment (pamgoodrich) [inline docs/guides/5-custom-functions.md]: You can probably remove "as much as possible"

File: docs/guides/5-custom-functions.md [hunk 10]
```diff
@@ -381,14 +384,14 @@ export default (targetVal, { excludedWords }) => {
...
 
-To keep your code cross-platform, you'd need to use a cross platform package such as [node-fetch](https://www.npmjs.com/package/node-fetch) or [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch), both of which implement spec-compliant fetch API and work in Node.js.
+To keep your code cross-platform, you'd need to use a cross-platform package such as [node-fetch](https://www.npmjs.com/package/node-fetch) or [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch), both of which implement spec-compliant fetch API and work in Node.js.
 
 ## Code Transpilation
 
 We encourage you to not transpile the code to ES5 if you can help it. Spectral does not support older environments than ES2019, so there is no need to bloat the bundle with useless transformations and polyfills. Ship untransformed async/await, do not include unneeded shims, it's all good.
 
-Prior to 6.x, Spectral hadn't supported ES Modules, yet as of recently using ES Modules is the recommended way to do things.
+Before 6.x, Spectral hadn't supported ES Modules, yet as of recently using ES Modules is the recommended way to do things.
```
Comment (pamgoodrich) [inline docs/guides/5-custom-functions.md:395]: I have no idea what version we are on now. Not sure we need this reference.

I see we have migration topics. I'd move any of this version-specific stuff to those topics. In general, the "current" doc should discuss things as they are now.

File: docs/reference/asyncapi-rules.md [hunk 11]
```diff
@@ -261,7 +261,7 @@ Operation objects should have a description.
 
 ### asyncapi-operation-operationId-uniqueness
 
-`operationId` must be unique across all the operations (except these one defined in the components).
+`operationId` must be unique across all the operations (except the ones defined in the components).
```
Comment (pamgoodrich) [inline docs/reference/asyncapi-rules.md:264]: This is the one that caught my eye after the latest updates. Thanks for fixing it!

File: docs/reference/openapi-rules.md [hunk 12]
```diff
@@ -382,8 +382,8 @@ Don't put query string items in the path, they belong in parameters with `in: qu
 
 Path parameters are correct and valid.
 
-1. For every parameters referenced in the path string (i.e: `/users/{userId}`), the parameter must be defined in either
-   `path.parameters`, or `operation.parameters` objects (Non standard HTTP operations will be silently ignored.)
+1. For every parameter referenced in the path string (i.e: `/users/{userId}`), the parameter must be defined in either
+   `path.parameters`, or `operation.parameters` objects (Non-standard HTTP operations will be silently ignored.)
```
Comment (pamgoodrich) [inline docs/reference/openapi-rules.md]: Non-standard > non-standard

**Review by pamgoodrich — DISMISSED** [review]

File: docs/guides/2-cli.md (hunk 3 above)
Comment (heitortsergent) [reply docs/guides/2-cli.md:65]: I'm not sure... I just looked at npm download numbers for v5 and the numbers are really small ([here](https://www.npmjs.com/package/@stoplight/spectral) and [here](https://www.npmjs.com/package/@stoplight/spectral-cli)), but I'm not sure if that's the only place to check.

@philsturgeon or @P0lip what do you think?

File: docs/guides/5-custom-functions.md (hunk 10 above)
Comment (heitortsergent) [reply docs/guides/5-custom-functions.md:395]: We're on v6 now, I think depending on the answer to the other comment we can update this to "ES Modules is the recommended way to do things.", or maybe remove it?

File: docs/reference/openapi-rules.md (hunk 12 above)
Comment (heitortsergent) [reply docs/reference/openapi-rules.md]: Thanks for catching that! I completely missed it 🤦

**Comment by heitortsergent** [comment]
@philsturgeon @P0lip @mnaumanali94 would any of you like to take a look at this? This ballooned into a big PR (sorry about that 😵‍💫), but most of the changes are grammar fixes.

**Review by philsturgeon — APPROVED** [review]
English Simplified (🇺🇸) > English Traditional (🇬🇧).

**Pull request merged**